generators/generator-base.js

```diff
@@ -1199,12 +1199,23 @@ module.exports = class extends PrivateBase {
      * @return {object} the composed generator
      */
     composeExternalModule(npmPackageName, subGen, options) {
-        const generatorName = jhipsterUtils.packageNameToNamespace(npmPackageName);
-        const generatorCallback = `${generatorName}:${subGen}`;
-        if (!this.env.get(generatorCallback)) {
-            throw new Error(`Generator ${generatorCallback} isn't registered.`);
+        try {
+            let generatorTocall = path.join(process.cwd(), 'node_modules', npmPackageName, 'generators', subGen);
+            if (!fs.existsSync(generatorTocall)) {
+                this.debug('using global module as local version could not be found in node_modules');
+                generatorTocall = path.join(npmPackageName, 'generators', subGen);
```

Comment:
I suspect that the generator could also be in 'lib/generators'. (I my case I have a generator written in typescript and hence the final JS is compiled to 'lib/generators'.)